src/sparseml/modifiers/obcq/utils/layer_compressor.py

```diff
@@ -0,0 +1,259 @@
+
+        self.inputs = None
+        torch.cuda.empty_cache()
+        return {"outputs": outputs}
+
+    def sequentially_compress(self, **kwargs):
+        """
+        Run compression module by module, in dependency order
+        """
+        subset = self.compressible_modules()
+
+        forward_args_spec = inspect.getfullargspec(self.layer.__class__.forward)
+        passed_in_args = [arg for arg in forward_args_spec.args if arg in kwargs]
+
+        passed_in_kwargs = {}
+        for arg in passed_in_args:
+            if isinstance(kwargs[arg], List):
+                passed_in_kwargs[arg] = kwargs[arg][0]
+            else:
+                passed_in_kwargs[arg] = kwargs[arg]
+        order = _find_dependency_order(
+            self.layer, subset, self.inputs[0], **passed_in_kwargs
+        )
+
+        nsamples = len(self.inputs)
+        for name in order:
+            gpts = SparseGPT(subset[name])
+
+            def add_batch(name):
+                def tmp(_, inp, out):
+                    gpts.add_batch(inp[0].data, out.data)
+
+                return tmp
+
+            handle = subset[name].register_forward_hook(add_batch(name))
+            for j in range(nsamples):
+                passed_in_kwargs = {}
+                for arg in passed_in_args:
+                    if isinstance(kwargs[arg], List):
+                        passed_in_kwargs[arg] = kwargs[arg][j]
+                    else:
+                        passed_in_kwargs[arg] = kwargs[arg]
+                self.layer(self.inputs[j], **passed_in_kwargs)
+            handle.remove()
+
+            _LOGGER.info(f"Compressing module {name} of layer {self.layer_index}")
+            gpts.fasterprune(
+                self.args["sparsity"],
+                prunen=self.args["prunen"],
+                prunem=self.args["prunem"],
+                percdamp=self.args["percdamp"],
+                blocksize=self.args["blocksize"],
+            )
+            gpts.free()
+
+
+def _find_dependency_order(layer, subset, an_input, **kwargs):
+    order = []
+
+    def exe_input(name):
+        def _exe_input(_, inp, out):
+            if name in subset:
+                order.append(name)
+
+        return _exe_input
+
+    handles = [subset[name].register_forward_hook(exe_input(name)) for name in subset]
+    layer(an_input, **kwargs)
+    for h in handles:
+        h.remove()
+    return order
+
+
+def _find_quant_layers(module, layers=[torch.nn.qat.Linear], name=""):
+    if type(module) in layers:
+        pieces = name.split(".")
+        if pieces[-1] == "module":
+            name = ".".join(pieces[:-1])
+        return {name: module}
+    res = {}
+    for name1, child in module.named_children():
+        res.update(
+            _find_layers(
+                child, layers=layers, name=name + "." + name1 if name != "" else name1
+            )
+        )
+    return res
+
+
+def _find_layers(module, layers=[nn.Conv2d, nn.Linear], name=""):
```

Comment:
curious about defaults here with the first supporting qat Linear but second Conv2d and Linear as well by default.

could also use comment on the relationship and usage of these two